rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -58,7 +58,7 @@ where
         Arc<base_client_gen::SetNotificationPreferenceForDeviceUpdatesCommandInvoker<C>>,
     create_or_update_discovered_device_command_invoker:
         Arc<discovery_client_gen::CreateOrUpdateDiscoveredDeviceCommandInvoker<C>>,
-    device_update_notification_dispatcher: Arc<Dispatcher<(Device, Option<AckToken>)>>,
+    device_update_notification_dispatcher: Arc<Dispatcher<(Device, Option<AckToken>), String>>,
```

Comment:
wasn't the whole goal of the dispatcher change to be able to use the DeviceEndpointRef and DeviceEndpointAssetRef as the hashmap key instead of using the hash function to a String?